### PR TITLE
[GPU] Add weight size limit for conv opt kernels

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Add weight size limit for conv opt kernels*

### Tickets:
 - *135788*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_os_iyx_osv16.cpp [hunk 1]
```diff
@@ -205,6 +205,14 @@ bool ConvolutionKernel_bfyx_os_iyx_osv16::Validate(const Params& p) const {
...
     }
 
+    // To prevent big sized filter which causes lots of CL build time.
+    const size_t acceptalbe_filter_size = 1024;     // This acceptable size was decided by heuristics
+    const auto& params = static_cast<const convolution_params&>(p);
+    auto filter_size = params.filterSize.x * params.filterSize.y;
+    if (filter_size > acceptalbe_filter_size) {
+        return false;
+    }
+
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_os_iyx_osv16.cpp:215]: let me repeat my question here: does customer need this change? for what?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_gemm_like.cpp [hunk 2]
```diff
@@ -110,6 +110,12 @@ bool ConvolutionKernel_bfyx_GEMMLike::Validate(const Params& p) const {
         return false;
     }
 
+    // To prevent big sized filter which causes lots of CL build time.
+    const size_t acceptalbe_filter_x_size = 64;     // This acceptable size was decided by heuristics
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_gemm_like.cpp:114]: nit: `acceptalbe` -> `acceptable`

Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_gemm_like.cpp:114]: why the condition does not include y-axis? why is it different from the condition from `convolution_kernel_bfyx_os_iyx_osv16.cpp`?

Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_gemm_like.cpp]: Fixed

Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_bfyx_gemm_like.cpp]: convolution_kernel_bfyx_gemm_like kernel creates LOOP jit constants by filter's x-axis value.

**Review by wilson-seok — APPROVED** [review]

**Review by hyunback — APPROVED** [review]

**Pull request merged**